docs/admin/administration.md

```diff
@@ -960,3 +960,28 @@ DAOS v2.2 client connections to pools which were created by DAOS v2.4
 will be rejected. DAOS v2.4 client should work with DAOS v2.4 and DAOS v2.2
 server. To upgrade all pools to latest format after software upgrade, run
 `dmg pool upgrade <pool>`
+
+### Interoperability Matrix
+
+The following table is intended to visually depict the interoperability
+policies for all major components in a DAOS system.
+
+
+||Server<br>(daos_server)|Engine<br>(daos_engine)|Agent<br>(daos_agent)|Client<br>(libdaos)|Admin<br>(dmg)|
+|:---|:---:|:---:|:---:|:---:|:---:|
+|Server|x.y|x.y.z|x.(y±1)|n/a|x.y|
```

Comment:
libdaos client <-> engine interoperability in this table shows minor version differences of at most 1.
However in this same document in the Software Upgrade section it states that DAOS v2.0 will maintain RPC protocol and persistent format compatibility for all future DAOS 2.x versions.

> DAOS v2.0 will maintain interoperability for both the wire protocol and persistent layout with any future v2.x versions. That being said, it is required that all engines in the same system run the same DAOS version.

Note: In some upcoming work for 2.6 feature development I will be needing to update the client <-> engine RPC protocol versions, and the pool metadata layout version too. While doing this I will need to know whether back-compatibility all the way back to 2.2 (and 2.0) software is required (I am assuming it is). If my assumption is correct then it may require libdaos/engine managing not just two protocol versions (current and "minus 1"), but also a third version, something not yet done in the development.
That other work is described in https://daosio.atlassian.net/wiki/spaces/DAOS/pages/11411390465/Metadata+Duplicate+Operation+Detection and I have an (admittedly) imperfect description of interop challenges in that document.

Reply:
may want feedback from @wangshilong on this